Source/WebCore/loader/HTTPHeaderField.cpp

```diff
@@ -30,16 +30,31 @@
 
 namespace WebCore {
 
-std::optional<HTTPHeaderField> HTTPHeaderField::create(String&& unparsedName, String&& unparsedValue)
+HTTPHeaderField::HTTPHeaderField(String&& name, String&& value)
+{
+    StringView strippedName = StringView(name).stripLeadingAndTrailingMatchedCharacters(RFC7230::isWhitespace<UChar>);
+    StringView strippedValue = StringView(value).stripLeadingAndTrailingMatchedCharacters(RFC7230::isWhitespace<UChar>);
+    
+    ASSERT(RFC7230::isValidName(strippedName) && RFC7230::isValidValue(strippedValue));
+
+    m_name = strippedName.length() == name.length() ? WTFMove(name) : strippedName.toString();
+    m_value = strippedValue.length() == value.length() ? WTFMove(value) : strippedValue.toString();
+}
+
+bool HTTPHeaderField::isValid(const String& name, const String& value)
 {
-    StringView strippedName = StringView(unparsedName).stripLeadingAndTrailingMatchedCharacters(RFC7230::isWhitespace<UChar>);
-    StringView strippedValue = StringView(unparsedValue).stripLeadingAndTrailingMatchedCharacters(RFC7230::isWhitespace<UChar>);
+    StringView strippedName = StringView(name).stripLeadingAndTrailingMatchedCharacters(RFC7230::isWhitespace<UChar>);
+    StringView strippedValue = StringView(value).stripLeadingAndTrailingMatchedCharacters(RFC7230::isWhitespace<UChar>);
     if (!RFC7230::isValidName(strippedName) || !RFC7230::isValidValue(strippedValue))
-        return std::nullopt;
+        return false;
+    return true;
+}
 
-    String name = strippedName.length() == unparsedName.length() ? WTFMove(unparsedName) : strippedName.toString();
-    String value = strippedValue.length() == unparsedValue.length() ? WTFMove(unparsedValue) : strippedValue.toString();
-    return {{ WTFMove(name), WTFMove(value) }};
+std::optional<HTTPHeaderField> HTTPHeaderField::create(String&& unparsedName, String&& unparsedValue)
```

Comment:
I think we should use CreateUsing=create and make a new attribute to allow the create function to return a std::optional, then just leave HTTPHeaderField class alone.